core/src/test/java/org/apache/iceberg/TestMetrics.java

```diff
@@ -270,6 +276,141 @@ public void testMetricsForNullColumns() throws IOException {
     assertBounds(1, IntegerType.get(), null, null, metrics);
   }
 
+  @Test
+  public void testMetricsForTopLevelWithMultipleRG() throws Exception {
+    Schema schema = getSimpleSchema();
+    int minimumRowGroupRecordCount = 100;
+    List<GenericData.Record> records = new ArrayList<>(201);
+
+    GenericData.Record firstRecord = new GenericData.Record(AvroSchemaUtil.convert(schema.asStruct()));
+    firstRecord.put("booleanCol", true);
+    firstRecord.put("intCol", 1);
+    // adding 1st element as null
+    firstRecord.put("longCol", null);
+    firstRecord.put("floatCol", 1.0F);
+    // adding 1st element as null
+    firstRecord.put("doubleCol", null);
+    // adding 1st element as null
+    firstRecord.put("decimalCol", null);
+    firstRecord.put("stringCol", "AAA");
+    firstRecord.put("dateCol", 1);
+    firstRecord.put("timeCol", 1L);
+    firstRecord.put("timestampCol", 1L);
+    firstRecord.put("uuidCol", uuid);
+    firstRecord.put("fixedCol", fixed);
+    firstRecord.put("binaryCol", "S".getBytes());
+    records.add(firstRecord);
```

Comment:
Why is firstRecord needed? To add nulls, why not use `i % 20 == 0 ? null : i`?

Reply:
refactored.

Actually initially I use ternary operator for multiple lines, and that result into cyclo-matic complexity = 14 , which is suppose to be 12 as per our checkstyle rule.
So, to keep it simple added first record.